controllers/selfnoderemediation_controller.go

```diff
@@ -161,8 +169,33 @@ func (r *SelfNodeRemediationReconciler) Reconcile(ctx context.Context, req ctrl.
 	return result, r.updateSnrStatusLastError(snr, err)
 }
 
-func (r *SelfNodeRemediationReconciler) isFencingCompleted(snr *v1alpha1.SelfNodeRemediation) bool {
-	return snr.Status.Phase != nil && *snr.Status.Phase == fencingCompletedPhase && snr.DeletionTimestamp != nil
+func (r *SelfNodeRemediationReconciler) patchSnrStatus(changed, org *v1alpha1.SelfNodeRemediation) error {
+	if err := r.Client.Status().Patch(context.Background(), changed, client.MergeFrom(org)); err != nil {
+		r.logger.Error(err, "failed to patch SNR status")
+		return err
+	}
+	return nil
+}
+
+func (r *SelfNodeRemediationReconciler) updateSnrRemediationPhase(phase remediationPhase, snr *v1alpha1.SelfNodeRemediation) error {
+	org := snr.DeepCopy()
+	phaseString := string(phase)
+	snr.Status.Phase = &phaseString
+	if err := r.patchSnrStatus(snr, org); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *SelfNodeRemediationReconciler) getPhase(snr *v1alpha1.SelfNodeRemediation) remediationPhase {
+	if snr.Status.Phase == nil {
+		return fencingStartedPhase
+	}
+	return remediationPhase(*snr.Status.Phase)
```

Comment:
I think we should verify the phase is valid.
At the moment this will work for any string.